src/main/webapp/WEB-INF/resources/sample-data/course.csv

```diff
@@ -0,0 +1,4 @@
+key,code,key:lecturer,title,course-abstract,nature,retail-price,more-info,draft-mode
+course-01,A 001,lecturer-01,Introduction to Computer Science,"Learn algorithms, data structures, programming languages, and software engineering principles in this intro course.",balanced,EUR 9999.99,https://example.com/A-001,true

```

Comment:
I think these codes do not fit the pattern given in the requirement's document, plus I think you haven't placed the restraint @Pattern("") in the Courses entity.